### PR TITLE
Fix minor issues with tag system

### PR DESCRIPTION
**Pull request by alicetouhou**
I'll be honest, this is not my best code. But HTML behavior is extremely strange sometimes, and at least this definitely works.

- Deck List updates properly when duplicating, importing, or deleting decks
- Tag filter works properly when loading into the screen
- Tag color doesn't change back to white after creating a tag. This was intended, but I'm not really sure why I decided to do this. Anyway, the old way I implemented it was wrong anyway so I fixed it.
- The deck screen ALWAYS scrolls to the correct position, so the deck you selected is always visible. New feature, but this fixes some issues too.

Would appreciate a review. This is very spaghetti

### CODE REVIEW

**Comment by alicetouhou** [comment]
Also, not in the scope of this PR, but we should eventually work on visual improvements for the tags. its not great. possibly @DevChimera can help out here

**Comment by Sense101** [comment]
Huh good thing I didn't decide to work on this in the end.
I will review this when I get a chance.

And I'd be happy to work on visual improvements, I noticed a few things already. What specific things do you want improved?

**Comment by alicetouhou** [comment]
> And I'd be happy to work on visual improvements, I noticed a few things already. What specific things do you want improved?

I just think my design sucks. Want to see what other people can come up with and improve from there

**Comment by alicetouhou** [comment]
Let's make this an issue though if we're gonna discuss

File: client/src/app/deck/deck-select.tsx [hunk 1]
```diff
@@ -194,8 +208,15 @@ function SelectDeck({
...
-		deckExists && setShowOverwriteModal(true)
-		!deckExists && saveDeckInternal(deck)
+		if (deckExists) {
+			setShowOverwriteModal(true)
+			return
+		}
+
+		saveDeckInternal(deck)
+		setSavedDecks(getSavedDecks())
+		setSortedDecks(sortDecks([...savedDecks, JSON.stringify(deck)]))
```
Comment (Lunarmagpie) [inline client/src/app/deck/deck-select.tsx:218]: Why are you stringifying then parsing later?

Comment (alicetouhou) [reply client/src/app/deck/deck-select.tsx:218]: The function takes in only strings... YEah this is really bad. I should refactor this part of the code so multiple behavior isn't handled in one function

Comment (Lunarmagpie) [reply client/src/app/deck/deck-select.tsx:218]: Please fix this... this function should not take a string.

Comment (alicetouhou) [reply client/src/app/deck/deck-select.tsx]: Resolved

**Comment by Lunarmagpie** [comment]
LGTM

**Review by Lunarmagpie — APPROVED** [review]

**Comment by alicetouhou** [comment]
That one change actually helped clean up a lot of the code. The only issue I have now is I need to set multiple states the same thing and I have no idea why.

**Comment by Lunarmagpie** [comment]
You should fix that

File: client/src/app/deck/deck-select.tsx [hunk 2]
```diff
@@ -395,7 +442,9 @@ function SelectDeck({
 			<AlertModal
 				setOpen={showDuplicateDeckModal}
 				onClose={() => setShowDuplicateDeckModal(!showDuplicateDeckModal)}
-				action={() => duplicateDeck(loadedDeck)}
+				action={() => {
+					duplicateDeck(loadedDeck)
+				}}
```
Comment (Sense101) [inline client/src/app/deck/deck-select.tsx:447]: Do we need these extra brackets here?

File: client/src/app/deck/deck-select.tsx [hunk 3]
```diff
@@ -61,12 +61,14 @@ function SelectDeck({
 
 	// STATE
 	const [savedDecks, setSavedDecks] = useState<Array<string>>(getSavedDecks)
-	const sortedDecks = savedDecks
-		.map((d: any) => {
+	function parseDecks(decks: string[]): Array<PlayerDeckT> {
```
Comment (Sense101) [inline client/src/app/deck/deck-select.tsx:64]: This seems extremely overcomplicated to me. We're saving the decks as strings and continuously reparsing them. What should really be happening is we should have a tracked state for a PlayerDeckT array that gets updated using the useState stuff, and every time a change is made we should stringify and save decks to local storage. We should not need to parse from strings other than when we first load.

This was probably always set up wrong tbh. This whole thing could be simplified a lot, but right now it works so therefore I think we should just merge it, and then redo the whole file at a later date.

**Review by Sense101 — COMMENTED** [review]
Reviewed. It's very spaghetti but the file was already such long spaghetti I don't think it matters. See review comments for more detail.

File: client/src/app/deck/deck-edit.tsx [hunk 4]
```diff
@@ -128,7 +128,7 @@ const addTag = (
 	if (tags.length >= 3) return
 	if (tag.name.length === 0) return
 	setTags([...tags, tag])
-	setColor('ffffff')
+	setColor(color)
```
[the remark this answers is not in the document]
Comment (alicetouhou) [reply client/src/app/deck/deck-edit.tsx:131]: I don't think this is in scope as I didn't modify the actual tag system with this PR, just a bug fix. But we can redo it in the future when I have time

**Pull request merged**